parquet-column/src/main/java/org/apache/parquet/io/RecordReader.java

```diff
@@ -40,4 +42,30 @@
   public boolean shouldSkipCurrentRecord() {
     return false;
   }
+
+  /**
+   * Reads a vector of rows into the given vectors.
+   * This method is for reading Parquet primitive values and
+   * does not call the converters
+   * @param vectors the vectors to fill
+   * @param columnSchemas schemas of the columns in the column vectors
+   * @param current number of records read so far
+   * @param total total number of records in the row group
+   */
+  public void readVectors(ColumnVector[] vectors, MessageType[] columnSchemas, long current, long total) {
```

Comment:
please create a class for { ColumnVector; MessageType } in cases like this. where you have pairs of things instead of co-indexed arrays.

But anyway, this should probably be in a different class. How about creating a VectorizedRecordReader ? It is already supposed to know the types of the columns.